docs/attributes-registry/vulnerability.md

```diff
@@ -0,0 +1,43 @@
+| `vulnerability.scanner.vendor`      | string   | The name of the scanner vendor that identified the vulnerability.                                                                                                                                                                                                                                                            | `Tenable`                                                                         | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `vulnerability.score.base`          | double   | The base score of the vulnerability. [1]                                                                                                                                                                                                                                                                                     | `7.5`                                                                             | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `vulnerability.score.environmental` | double   | The environmental score of the vulnerability. [2]                                                                                                                                                                                                                                                                            | `7.5`                                                                             | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `vulnerability.score.temporal`      | double   | The temporal score of the vulnerability. [3]                                                                                                                                                                                                                                                                                 | `7.5`                                                                             | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `vulnerability.score.version`       | string   | The version of the scoring system used to calculate the vulnerability score.                                                                                                                                                                                                                                                 | `3.0`                                                                             | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
My only comment here is whether or not it makes sense to have some of these be a `CVSS` specific attribute, particularly when I see things like score ranging from 0.0 to 10.0.

I like that this is based on the CVSS specification, what I can't tell as an outsider, what other standards can be mapped here.  I'm waiting for more security folks to chime in on these to understand if this does, indeed, match expectations for how to communicate vulnerabilities via Events, etc.

Reply:
@jsuereth  - agreed. If using numbers this should be clearly defined (ie. CVSS).

Also, what examples of "score.version" exist? The other examples in this file assume CVSS (like score), but the classification attribute technically allows for others which may not 1:1 align.

Reply:
I'd agree that there could be some challenges in having the `score.version` allow for other systems. Some of the fields we've got here are specific to CVSS scoring (e.g. environmental and temporal scores). If the intent is to allow for other scoring regimes (and I do think that makes sense), I'd be inclined to have a `vulnerability.score` value and not have the other CVSS specific items.

It's also worth noting that generally scanners can't/shouldn't provide values for environmental and temporal metrics as those require information about the working environment of the impacted system that scanners often don't have.

One example of the type of score someone might want to add is EPSS (https://www.first.org/epss/model) which expresses a score as a percentage.

Reply:
@jsuereth and folks I agree with the idea that we must allow other standards to be taken into account though CVSS is probably the most widely used. We do have also notation about CVSS in comments. 

I see here 2 ways of solving it

**A**

- introducing `score.standard` field, which could be used to define the actual standard.
- `score.version` and `score.base` will remain as is as common fields for all standards
- if any standard requires more fields, those could be added with sub-namespacing in the beginning such `score.cvss.temporal` etc
- as of now we will add only CVSS but the schema could be extended if needed

**B**
- use every standard under it's own sub-namespace. i.e. current `score` fields all go under `score.cvss.*`
- that way we will not have to define `standard` additionally. 

I prefer the first option as it gives more flexibility in using any standard without updating schema, if we care only about `score.base`. But this will make `base.score` definition a bit vague, because different standards can define it using different measurements: 0.0 to 1.0, 0.0 to 10.0, 0.0 to 100.0 etc. But we can add it to the notes

Reply:
Yep that sounds like a good idea to me, Option A you've outlined there should help clarify this area.

Reply:
I believe there are two things that need to be considered:

1. The severity/score reported by the vendor that found this vulnerability in a specific artifact.
2. The various scores this vulnerability may get from various standards

A decision to make: Are we defining a vulnerability object that can later be referenced from a finding, or are we defining a vulnerability finding reported by a vendor on a vulnerable artifact?

Depending on the purpose, it could go in one or another direction.
For 1. - we would need to have a separate severity/score for the finding, and potentially an array of scores defined by various standards.
For 2. - we would still need a score used by the vendor as the base score, and potentially an array of additional scores that were considered by that vendor.

See t[his example of Snyk](https://security.snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-559094):
* They talk about a vulnerability, define their own score, and reference multiple standards.
* When they will run the assessment of an artifact against this vulnerability, they will generate a finding event, with the finding score adjusted to additional contextual information.
* 
![image](https://github.com/user-attachments/assets/3401781d-c095-4d62-b13b-37ce60267d83)